packages/cadl-ranch-specs/http/routes/main.tsp

```diff
@@ -0,0 +1,476 @@
+      Value: "a"
+      Expected path: /routes/path/annotation-only/a
+    """)
+  @route("annotation-only")
+  op annotationOnly(@path param: string): void;
+
+  @route("reserved-expansion")
+  namespace ReservedExpansion {
+    @scenario
+    @scenarioDoc("""
+      Defines a path parameter that shouldn't encode reserved characters. It should however still encode the other url characters.
+        Param value: "foo/bar baz"
+        Expected path: "/routes/path/reserved-expansion/template/foo/bar%20baz"
+      """)
+    @route("template/{+param}")
+    op template(param: string): void;
+
+    @scenario
+    @scenarioDoc("""
+      Defines a path parameter that shouldn't encode reserved characters. It should however still encode the other url characters.
+        Param value: "foo/bar baz"
+        Expected path: "/routes/path/reserved-expansion/annotation/foo/bar%20baz"
+      """)
+    @route("annotation")
+    op annotation(@path(#{ allowReserved: true }) param: string): void;
+  }
+
+  @route("simple")
+  namespace Simple {
+    @route("standard")
+    namespace Standard {
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: false when passed a primitive value.
+            Param value: "a"
+            Expected path: /routes/simple/standard/primitivea
+        """)
+      @route("primitive{param}")
+      op primitive(param: string): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: false when passed an array value.
+            Param value: ["a","b"]
+            Expected path: /routes/simple/standard/arraya,b
+        """)
+      @route("array{param}")
+      op array(param: string[]): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: false when passed a record value.
+            Param value: {a: 1, b: 2}
+            Expected path: /routes/simple/standard/recorda,1,b,2
+        """)
+      @route("record{param}")
+      op record(param: string[]): void;
+    }
+
+    @route("explode")
+    namespace Explode {
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: true when passed a primitive value.
+            Param value: "a"
+            Expected path: /routes/simple/explode/primitivea
+        """)
+      @route("primitive{param*}")
+      op primitive(param: string): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: true when passed an array value.
+            Param value: ["a","b"]
+            Expected path: /routes/simple/explode/arraya.b
+        """)
+      @route("array{param*}")
+      op array(param: string[]): void;
+
+      @scenario
+      @scenarioDoc("""
+        Test simple expansion with explode: true when passed a record value.
+            Param value: {a: 1, b: 2}
+            Expected path: /routes/simple/explode/recorda=1,b=2
+        """)
+      @route("record{param*}")
+      op record(param: string[]): void;
```

Comment:
How's param value `{a: 1, b: 2}` (looks like a dict) matches the op `op record(param: string[])` (`param` an array)?